agent/rpc/client_grpc.go

```diff
@@ -277,17 +271,16 @@ func (c *client) Extend(ctx context.Context, id string) (err error) {
 }
 
 // Update updates the workflow state.
-func (c *client) Update(ctx context.Context, id string, state rpc.State) (err error) {
+func (c *client) Update(ctx context.Context, id string, state rpc.StepState) (err error) {
 	retry := c.newBackOff()
 	req := new(proto.UpdateRequest)
 	req.Id = id
-	req.State = new(proto.State)
-	req.State.Error = state.Error
-	req.State.ExitCode = int32(state.ExitCode)
-	req.State.Exited = state.Exited
-	req.State.Finished = state.Finished
-	req.State.Started = state.Started
+	req.State = new(proto.StepState)
 	req.State.StepUuid = state.StepUUID
+	req.State.Started = state.Started
```

Comment:
Missing exited

Reply:
Do you have an idea what exited is actually used for? Could we just use `Finished != 0`?

Reply:
well yes but I would rather have it removed in an dedicated pull to be able to git-bisect if things go wrong ...

Reply:
Readded it for now